multiqc/modules/checkqc/checkqc.py

```diff
@@ -0,0 +1,571 @@
+            if len(self.log_files) == 1:
+                sample = self.clean_s_name(f"{lane} - {read}", f)
+            else:
+                sample = self.clean_s_name(f"{lane} - {read} ({run})", f)
+
+            if self.is_ignore_sample(sample):
+                continue
+            self.add_data_source(f, sample)
+            error_rate = issue["data"]["error_rate"]
+            threshold = issue["data"]["threshold"]
+            data[sample] = {"threshold": threshold}
+            if is_error:
+                data[sample]["missing_error"] = error_rate - threshold
+            else:
+                data[sample]["missing_warning"] = error_rate - threshold
+        if data:
+            if "ErrorRateHandler" not in self.checkqc_data:
+                self.checkqc_data["ErrorRateHandler"] = data
+            else:
+                self.checkqc_data["ErrorRateHandler"].update(data)
+
+    def add_error_rate_section(self):
+        """Add a section for lanes with error rate too high
+
+        Creates a barplot with error rate and missing to reach threshold
+
+        """
+        data = self.checkqc_data["ErrorRateHandler"]
+
+        warning, error = self._get_warning_error(data)
+
+        cats = OrderedDict()
+        cats["threshold"] = {
+            "name": "Error rate part until threshold",
+        }
+        if warning:
+            cats["missing_warning"] = {"name": "Error rate part above threshold for QC warning", "color": "#ffc300"}
+        if error:
+            cats["missing_error"] = {"name": "Error rate part above threshold for QC error", "color": "#ff0000"}
+
+        pconfig = {
+            "id": "checkqc_error-rate-plot",
+            "title": "CheckQC: Error rate too high",
+            "ylab": "Error rate",
+            "xlab": "Lane - Read",
+            "cpswitch": False,
+        }
+
+        self.add_section(
+            name="Error rate too high",
+            anchor="checkqc-errorrate",
+            description="Some lanes have too high error rate.",
+            plot=bargraph.plot(data, cats, pconfig),
+        )
+
+    def get_undetermined_percentage_data(self, issues, run, f):
+        """Get data from UndeterminedPercentageHandler
+
+        Args:
+            issues (dict): JSON dict from CheckQC containing UndeterminedPercentageHandler results
+            run (str): name of sequencing run
+            f (dict): MultiQC log file dict
+        """
+        data = {}
+        for issue in issues:
+            is_error = issue["type"] == "error"
+            lane = str(issue["data"]["lane"])
+            if len(self.log_files) == 1:
+                sample = self.clean_s_name(lane, f)
+            else:
+                sample = self.clean_s_name(f"{lane} ({run})", f)
+
+            if self.is_ignore_sample(sample):
+                continue
+            self.add_data_source(f, sample)
+            p_undetermined = issue["data"]["percentage_undetermined"]
+            threshold = issue["data"]["threshold"]
```

Comment:
CheckQC [reports](https://github.com/Molmed/checkQC/blob/36070f4d0227fec42ec6001a6d46bc9db198a856/checkQC/handlers/undetermined_percentage_handler.py#L56) that percentage undetermined couldn't be computed if `yield == 0`, in that case `data` only contains `lane` and `percentage_underermined` (which is `N/A`).  It would be nice if that case was handled.

Reply:
Thanks for the pointer, fixed in ead255e, this case does not break parsing the checkqc JSON now. I added it to the plot as an empty entry so the user can at least see that something is up with the respective sample. Not sure how to transport more info, e.g. that yield is zero, because no legend is shown for a bar of zero length.

Reply:
![checkqc_module_no_yield](https://user-images.githubusercontent.com/1584541/143001318-47b7ca45-9e9d-4465-9ffa-b33eeb380df5.png)
I tried it out and it looks like this. I think this is fine, but it would nice if the message you wrote ("Yield is 0, no undetermined percentage computation possible") was communicated. I'm not super familiar with how MultiQC modules work, so can't really suggest how to best implement it. Would it be helpful if I send you the json file I used?

Reply:
In your test, was the yield 0 in all lanes? I had not thought of this possibility. The corresponding JSON file would be helpful, thanks! Below is a plot where only one lane has yield 0 ( (lane 8 on run hiseq2500_rapidhighoutput_v4_1), I was hoping the legend would include the "Yield is 0" message, but it does not for a value of 0.
I have to see how to do this best, I was hoping to somehow include it in the barplot, but probably a separate plot or section is needed. 

![checkqc_undetermined-percentage-plot](https://user-images.githubusercontent.com/3832576/143006025-78dc3a6b-01f0-4a8f-b1f6-3ef632777589.png)

Reply:
Yeah, It's basically a failed MiSeq run, so only one lane. It's quite clear from the other handlers that something is wrong, so I don't see it as super necessary to fix, just a nice to have. I can't upload the file here, so send me an e-mail (you can find my address on my profile) if you want me to send the file. But like I said, I'm also fine with approving this as is.

Reply:
I created a dummy JSON with an entry like this for testing, I think that should cover it (correct me if I'm wrong):

```json
    "UndeterminedPercentageHandler": [
        {
            "type": "error",
            "message": "The percentage of undetermined indexes was to high on lane 8, it was: N/A",
            "data": {
              "lane": 1,
              "percentage_undetermined": "N/A"
            }
        }
    ],
```

In 1d28323 I added an additional section for cases like this, it contains a table listing the lanes with yield zero.

Reply:
Yeah, exactly, only the message is a bit different, but it isn't used in the code as far as I can see.
```
    "UndeterminedPercentageHandler": [
        {
            "type": "error",
            "message": "Yield for lane: {} was 0. No undetermined percentage could be computed.",
            "data": {
                "lane": 1,
                "percentage_undetermined": "N/A"
            }
        }
    ],
```
(Oops, we have a :bug: {})